src/components/Hoverable/index.tsx

```diff
@@ -1,213 +1,29 @@
-    }, [isHovered]);
-
-    useEffect(() => {
-        if (!disabled || !isHovered) {
-            return;
-        }
-        setIsHovered(false);
-    }, [disabled, isHovered]);
-
-    useEffect(() => {
-        if (disabled) {
-            return;
-        }
-        if (onHoverIn && isHovered) {
-            return onHoverIn();
-        }
-        if (onHoverOut && !isHovered) {
-            return onHoverOut();
-        }
-    }, [disabled, isHovered, onHoverIn, onHoverOut]);
-
-    // Expose inner ref to parent through outerRef. This enable us to use ref both in parent and child.
-    useImperativeHandle<HTMLElement | null, HTMLElement | null>(outerRef, () => ref.current, []);
-
-    const child = useMemo(() => React.Children.only(mapChildren(children as ReactElement, isHovered)), [children, isHovered]);
-
-    const enableHoveredOnMouseEnter = useCallback(
-        (event: MouseEvent) => {
-            updateIsHoveredOnScrolling(true);
-            onMouseEnter(event);
-
-            if (typeof child.props.onMouseEnter === 'function') {
-                child.props.onMouseEnter(event);
-            }
-        },
-        [child.props, onMouseEnter, updateIsHoveredOnScrolling],
-    );
-
-    const disableHoveredOnMouseLeave = useCallback(
-        (event: MouseEvent) => {
-            updateIsHoveredOnScrolling(false);
-            onMouseLeave(event);
-
-            if (typeof child.props.onMouseLeave === 'function') {
-                child.props.onMouseLeave(event);
-            }
-        },
-        [child.props, onMouseLeave, updateIsHoveredOnScrolling],
-    );
-
-    const disableHoveredOnBlur = useCallback(
-        (event: MouseEvent) => {
-            // Check if the blur event occurred due to clicking outside the element
-            // and the wrapperView contains the element that caused the blur and reset isHovered
-            if (!ref.current?.contains(event.target as Node) && !ref.current?.contains(event.relatedTarget as Node)) {
-                setIsHovered(false);
-            }
-
-            if (typeof child.props.onBlur === 'function') {
-                child.props.onBlur(event);
-            }
-        },
-        [child.props],
-    );
-
-    // We need to access the ref of a children from both parent and current component
-    // So we pass it to current ref and assign it once again to the child ref prop
-    const hijackRef = (el: HTMLElement) => {
-        ref.current = el;
-        if (child.ref) {
-            assignRef(child.ref, el);
-        }
-    };
-
-    if (!DeviceCapabilities.hasHoverSupport()) {
-        return React.cloneElement(child, {
-            ref: hijackRef,
-        });
+function Hoverable({isDisabled, ...props}: HoverableProps, ref: Ref<HTMLElement>) {
+    // If Hoverable is disabled, just render the child without additional logic or event listeners.
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    if (isDisabled || !hasHoverSupport()) {
```

Comment:
A quick note: You've updated the prop from `disabled` to `isDisabled` in this component, but it seems that the corresponding changes haven't been made where this component is being used

Reply:
[### ](https://github.com/margelo/expensify-app-fork/blob/e8dc64387e83b26a213785f9e51854f13f0c289f/src/pages/home/report/ReportActionItem.js#L675)

Reply:
https://github.com/margelo/expensify-app-fork/blob/e8dc64387e83b26a213785f9e51854f13f0c289f/src/pages/home/report/ReportActionItem.js#L675

Reply:
Also, I see this error when hovering 
<img width="1496" alt="image" src="https://github.com/Expensify/App/assets/48593211/58c8bdc2-5765-45e0-9c25-6ab055410f0c">

Reply:
due to this https://github.com/margelo/expensify-app-fork/blob/3895f80d2cdedf942197c8d60e221f9a54ab07fc/src/components/Tooltip/BaseTooltip/index.tsx#L192

Reply:
That's right, thanks for your feedback @perunt!

Reply:
I didn't dig deeper into the influence of that bug, but it might have crashed due to the context menu on the draft message. Just bump @mountiny to keep it on the radar

Reply:
Yeah, it looks like there was a mismatch between typing and actual usage. I've created a quickfix PR and notified @mountiny, thanks once again 👍

Reply:
Thanks for the quick response ❤️

Reply:
Merged, thank you ❤️